analysis/cross_partition_combiners.py

```diff
@@ -131,17 +131,19 @@ def _add_dataclasses_by_fields(dataclass1, dataclass2,
         if field.name in fields_to_ignore:
             continue
         value1 = getattr(dataclass1, field.name)
+        if value1 is None:
+            continue
         value2 = getattr(dataclass2, field.name)
         if dataclasses.is_dataclass(value1):
             _add_dataclasses_by_fields(value1, value2, fields_to_ignore)
             continue
         setattr(dataclass1, field.name, value1 + value2)
 
 
-def _multiply_float_dataclasses_field(dataclass, factor: float):
-    """Recursively multiply all float fields of the dataclass by given number.
+def _multiply_float_dataclasses_field(dataclass, factor: float) -> None:
```

Comment:
Where is this function supposed to be used?

Reply:
We compute the aggregate per dataset, i.e. sum per some fields. We'd like to return average, per field, this funciton will be used for averaging fields, i.e. multiplying by 1/N (it will be in the next PR :) )